### PR TITLE
Platform-3203 new log format

### PR DESCRIPTION
**Pull request by harnash**
This will allow gathering metrics from Traefik's access logs in both formats:
* common access log (combined)
* json formatted access log

This is configurable via annotations on the Traefik instances:
```js
"wikia_com/keys": "{\"es_logs\": {\"container_name\": \"traefik-ingress-controller\", \"es_index\": \"k8s-ingress-controller\"}, \"influx_metrics\": {\"container_name\": \"traefik-ingress-controller\", \"type\": \"access_logs_combined\"}}"
```
The "type key can have two values:
* access_logs_combined
* access_log_as_json

Additionally I've added config entry `Fields` which hold list of fields being exported to InfluxDB (see example config file for details).

### CODE REVIEW

File: metrics/traefik.go [hunk 1]
```diff
@@ -76,146 +87,164 @@ func NewTraefikMetricProcessor(config []common.RulesConfig) (*TraefikMetricProce
...
-	m := map[string]interface{}{}
-
-	// status codes
-	integer, err := strconv.ParseInt(values["status"], 10, 64)
+func parseCommonLog(entry model.LogEntry) (map[string]interface{}, error) {
+	matches := ApacheCombinedLogRegex.FindStringSubmatch(entry.Log)
 
-	if err != nil {
-		return nil, err
+	if len(matches) != 15 {
```
Comment (themech) [inline metrics/traefik.go]: `15` looks like a good candidate to move to a const

File: metrics/traefik.go [hunk 2]
```diff
@@ -76,146 +87,164 @@ func NewTraefikMetricProcessor(config []common.RulesConfig) (*TraefikMetricProce
...
 			"rule_id":       rule.Id,
 		}
 
-		pt, err := client.NewPoint(measurement, tags, values, time.Now())
+		values := map[string]interface{}{}
+
+		var timestamp time.Time
+		original_timestamp, has := parsedLog["start_utc"]
+		if has {
+			timestamp, err = time.Parse("", original_timestamp.(string))
```
Comment (themech) [inline metrics/traefik.go]: Why do we use the empty layout here?

**Review by themech — CHANGES_REQUESTED** [review]
Left two comments. Also did you consider adding some basic unit tests for `toSnake`, `parseCommonLog` and `parseJsonLog`?

File: metrics/traefik.go (hunk 2 above)
Comment (harnash) [reply metrics/traefik.go]: Good catch: 5bcc0f6a198137939f7af97c3f3434ba04f58011

File: metrics/traefik.go (hunk 1 above)
Comment (harnash) [reply metrics/traefik.go]: 5f45fef777762f9bf3c2828ed2e84a70e9c7e878

**Review by themech — APPROVED** [review]

**Pull request merged**